### PR TITLE
Create assertEquals for double with a delta parameter.

### PR DESCRIPTION
**Pull request by hugovdm**
Not knowing how to "test the test suite functions", I tested this in my local client with:

    double inf = std::numeric_limits<double>::infinity();
    double nan = std::numeric_limits<double>::quiet_NaN();
    // with delta
    assertEquals("should succeed 0", 2.70, 2.72, 0.02001);
    assertEquals("should succeed 1", inf, inf, 0.02);
    assertEquals("should succeed 2", nan, nan, 0.02);
    assertEquals("should succeed 3", nan, -nan, 0.02);
    assertEquals("should succeed 4", -nan, nan, 0.02);
    assertEquals("should fail 0", -inf, inf, 0.02);
    assertEquals("should fail 1", inf, -inf, 0.02);
    assertEquals("should fail 2", inf, 10000.0, inf);
    assertEquals("should fail 3", 10000.0, inf, inf);
    assertEquals("should fail 4", inf, 10000.0, nan);
    assertEquals("should fail 5", 1.0, 2.1, nan);
    assertEquals("should fail 6", nan, 10000.0, 0);
    assertEquals("should fail 7", 10000.0, nan, 0);
    // without delta
    assertEquals("should succeed 0", 2.70, 2.70);
    assertEquals("should succeed 1", inf, inf);
    assertEquals("should succeed 2", nan, nan);
    assertEquals("should succeed 3", nan, -nan);
    assertEquals("should succeed 4", -nan, nan);
    assertEquals("should fail 0", -inf, inf);
    assertEquals("should fail 1", inf, -inf);
    assertEquals("should fail 2", inf, 10000.0);
    assertEquals("should fail 3", 10000.0, inf);
    assertEquals("should fail 4", inf, 10000.0);
    assertEquals("should fail 5", 1.0, 2.1);
    assertEquals("should fail 6", nan, 10000.0);
    assertEquals("should fail 7", 10000.0, nan);

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-_____
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/intltest.h
Comment (sffc) [inline icu4c/source/test/intltest/intltest.h]: Is delta an absolute number, or a relative error?  It looks like an absolute number.  I would say "absolute delta" to clarify.

Comment (hugovdm) [reply icu4c/source/test/intltest/intltest.h]: Modified/Done, talking about "absolute difference".

"Absolute delta" suggests it might turn a negative into a positive. I'm not sure what the [JUnit assertEqual]( http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertEquals(java.lang.String,%20double,%20double,%20double)) does for negative numbers, the documentation seems to suggest it expects non-negative. (If we really want to emulate JUnit well, I should probably check?)

File: icu4c/source/test/intltest/intltest.cpp
Comment (hugovdm) [inline icu4c/source/test/intltest/intltest.cpp]: This comment block is to be deleted, if we want to stay as close to JUnit as possible. Otherwise, this CL is of very low importance.

File: icu4c/source/test/intltest/unitstest.cpp
Comment (sffc) [inline icu4c/source/test/intltest/unitstest.cpp]: This could also be `const char *const`

File: icu4c/source/test/intltest/unitstest.cpp
Comment (sffc) [inline icu4c/source/test/intltest/unitstest.cpp:92]: Very Optional: use `US_INV` constructor for UnicodeString.  For example, `UnicodeString(char*, -1, US_INV)`

File: icu4c/source/test/intltest/intltest.cpp
Comment (sffc) [inline icu4c/source/test/intltest/intltest.cpp]: Absolute difference is good here.

**Review by younies — COMMENTED** [review]
I think assertEqualsNear name is more expressive

File: icu4c/source/test/intltest/unitstest.cpp
Comment (hugovdm) [reply icu4c/source/test/intltest/unitstest.cpp:92]: Leaving it, for the sake of readability: I'm also only using UnicodeString because it has the nice operator+, generally a little bit wasteful...

File: icu4c/source/test/intltest/unitstest.cpp
Comment (hugovdm) [reply icu4c/source/test/intltest/unitstest.cpp]: I've changed this now, since I'm touching these lines anyway, but I'm leaving it a bit inconsistent: not touching the other similar examples in this file.

File: icu4c/source/test/intltest/intltest.cpp
Comment (hugovdm) [reply icu4c/source/test/intltest/intltest.cpp]: (Taking this as just an observation, not a suggested change.)

**Review by sffc — APPROVED** [review]

**Comment by hugovdm** [comment]
Thanks for the approve.

Experimentation regarding GitHub security branch protection measures:

- I did a rebase for doubleEqualsWithDelta (preserving the sequence of commits in this PR), and then tried:

$ git checkout units-staging
$ git reset --hard sffc/units-staging
$ git merge --no-ff doubleEqualsWithDelta
$ git diff 86b4db999f  # identical to the commit in this PR at which I got approval
$ git push sffc units-staging

That got rejected, protected branch:

remote: error: GH006: Protected branch update failed for refs/heads/units-staging.
remote: error: At least 1 approving review is required by reviewers with write access.

$ git checkout doubleEqualsWithDelta
$ git push --force

I see approval isn't lost. There's a GitHub option that would consider the approval stale, even if the contents are the same. (I wish there were an option that would maintain the approval, if the contents didn't change?)

I redid this, though it was actually unnecessary:
$ git checkout units-staging
$ git reset --hard sffc/units-staging
$ git merge --no-ff doubleEqualsWithDelta
$ git diff 86b4db999f
$ git push sffc units-staging

That worked... so GitHub only needed to see the connection to the branch as it was pushed first.

**Comment by hugovdm** [comment]
Also of note:

> hugovdm merged commit e92a2fc into sffc:units-staging
> 2 minutes ago
> 1 check failed
> * Travis CI - Pull Request Build Started

The "check failed" was due to not waiting for completion of the Travis run following the force-push. Results should succeed and the "check failed" should then be cleared, since the contents of 86b4db9 and 086881f are identical.

One can probably sneak in any number of commits after an approval, unless the related branch protection checkbox is checked.

**Pull request merged**
